From 5a5b2451f9e1c26ea09595ea2787b8362653d0c1 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 4 Nov 2020 12:21:59 +0100
Subject: [PATCH v2 2/2] riscv: enable SATA disk on QEMU RISC-V

Allow attaching a virtual SATA disk to QEMU RISC-V by implying
AHCI, AHCI_PCI, CMD_SCSI, DM_SCSI, PCI_INIT_R, SCSI, SCSI_AHCI.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
v2:
	add options to BOARD_SPECIFIC_OPTIONS instead of defconfig
---
 board/emulation/qemu-riscv/Kconfig | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/board/emulation/qemu-riscv/Kconfig b/board/emulation/qemu-riscv/Kconfig
index 36d3b68ab9..0818048ba6 100644
--- a/board/emulation/qemu-riscv/Kconfig
+++ b/board/emulation/qemu-riscv/Kconfig
@@ -29,12 +29,15 @@ config BOARD_SPECIFIC_OPTIONS # dummy
 	def_bool y
 	select GENERIC_RISCV
 	select SUPPORT_SPL
+	imply AHCI
 	imply SMP
 	imply BOARD_LATE_INIT
+	imply PCI_INIT_R
 	imply SPL_RAM_SUPPORT
 	imply SPL_RAM_DEVICE
 	imply CMD_PCI
 	imply CMD_POWEROFF
+	imply CMD_SCSI
 	imply CMD_PING
 	imply CMD_EXT2
 	imply CMD_EXT4
@@ -43,11 +46,15 @@ config BOARD_SPECIFIC_OPTIONS # dummy
 	imply DOS_PARTITION
 	imply ISO_PARTITION
 	imply EFI_PARTITION
+	imply SCSI_AHCI
+	imply AHCI_PCI
 	imply E1000
 	imply NVME
 	imply PCI
 	imply DM_PCI
 	imply PCIE_ECAM_GENERIC
+	imply SCSI
+	imply DM_SCSI
 	imply SYS_NS16550
 	imply SIFIVE_SERIAL
 	imply SYSRESET
-- 
2.28.0

